projects/rocprofiler-systems/source/lib/core/agent_info.hpp

```diff
@@ -0,0 +1,237 @@
+        bank["mem_clk_max"]           = agent_data.mem_banks[i].mem_clk_max;
+        bank["size_in_bytes"]         = agent_data.mem_banks[i].size_in_bytes;
+        data["mem_banks"].push_back(bank);
+    }
+
+    data["caches"] = nlohmann::json::array();
+    for(uint32_t i = 0; i < agent_data.caches_count; ++i)
+    {
+        nlohmann::json cache;
+        cache["processor_id_low"]    = agent_data.caches[i].processor_id_low;
+        cache["size"]                = agent_data.caches[i].size;
+        cache["level"]               = agent_data.caches[i].level;
+        cache["cache_line_size"]     = agent_data.caches[i].cache_line_size;
+        cache["cache_lines_per_tag"] = agent_data.caches[i].cache_lines_per_tag;
+        cache["association"]         = agent_data.caches[i].association;
+        cache["latency"]             = agent_data.caches[i].latency;
+        cache["type"]["Data"]        = agent_data.caches[i].type.ui32.Data;
+        cache["type"]["Instruction"] = agent_data.caches[i].type.ui32.Instruction;
+        cache["type"]["CPU"]         = agent_data.caches[i].type.ui32.CPU;
+        cache["type"]["HSACU"]       = agent_data.caches[i].type.ui32.HSACU;
+        data["caches"].push_back(cache);
+    }
+
+    data["io_links"] = nlohmann::json::array();
+    for(uint32_t i = 0; i < agent_data.io_links_count; ++i)
+    {
+        nlohmann::json link;
+        link["type"]          = agent_data.io_links[i].type;
+        link["version_major"] = agent_data.io_links[i].version_major;
+        link["version_minor"] = agent_data.io_links[i].version_minor;
+        link["node_from"]     = agent_data.io_links[i].node_from;
+        link["node_to"]       = agent_data.io_links[i].node_to;
+        link["weight"]        = agent_data.io_links[i].weight;
+        link["min_latency"]   = agent_data.io_links[i].min_latency;
+        link["max_latency"]   = agent_data.io_links[i].max_latency;
+        link["min_bandwidth"] = agent_data.io_links[i].min_bandwidth;
+        link["max_bandwidth"] = agent_data.io_links[i].max_bandwidth;
+        link["recommended_transfer_size"] =
+            agent_data.io_links[i].recommended_transfer_size;
+        link["flags"]["Override"]    = agent_data.io_links[i].flags.ui32.Override;
+        link["flags"]["NonCoherent"] = agent_data.io_links[i].flags.ui32.NonCoherent;
+        link["flags"]["NoAtomics32bit"] =
+            agent_data.io_links[i].flags.ui32.NoAtomics32bit;
+        link["flags"]["NoAtomics64bit"] =
+            agent_data.io_links[i].flags.ui32.NoAtomics64bit;
+        link["flags"]["NoPeerToPeerDMA"] =
+            agent_data.io_links[i].flags.ui32.NoPeerToPeerDMA;
+        data["io_links"].push_back(link);
+    }
+
+    data["gpu_index"] = (agent_data.type == ROCPROFILER_AGENT_TYPE_GPU)
+                            ? agent_data.logical_node_type_id
+                            : -1;
+
+    auto normalize_json_string = [](nlohmann::json& json_data) {
```

Comment:
Maybe a comment on this lambda.

```suggestion
    // Normalize JSON string by escaping quotes to prepare for SQL insertion
    auto normalize_json_string = [](nlohmann::json& json_data) {
```